Python/pywarpx/_libwarpx.py

```diff
@@ -594,11 +594,6 @@ def add_particles(self, species_name, x=None, y=None, z=None, ux=None, uy=None,
         for key, val in kwargs.items():
             assert np.size(val)==1 or len(val)==maxlen, f"Length of {key} doesn't match len of others"
 
-        # --- If the length of the input is zero, then quietly return
-        # --- This is not an error - it just means that no particles are to be injected.
-        if maxlen == 0:
-            return
-
```

Comment:
Is this causing us to diverge from upstream? What's the reasoning in removing it?

Reply:
Nope, the upstream PR to fix this was merged a little earlier as well. We need to remove it since there is a collective call the `Redistribute` at the end of `AddNParticles` that is not reached by some processors (if they inject no particles) due to this early return. That bug caused the simulation to hang.